go/vt/discovery/topology_watcher_test.go

```diff
@@ -410,3 +410,32 @@ func TestFilterByShard(t *testing.T) {
 		}
 	}
 }
+
+var testFilterByKeyspace = []struct {
+	keyspace string
+	expected bool
+}{
+	{"ks1", true},
+	{"ks2", true},
+	{"ks3", false},
+	{"ks4", true},
+	{"ks5", true},
+	{"ks6", false},
+	{"ks7", false},
+}
+
+func TestFilterByKeyspace(t *testing.T) {
+	selectedKeyspaces := []string{"ks1", "ks2", "ks4", "ks5"}
+	fbk := NewFilterByKeyspace(nil, selectedKeyspaces)
```

Comment:
For completeness and If there is a tablet recorder for unit tests, I think it will be good to use it here. Then you can add tests for `AddTablet`, `RemoveTablet` and `ReplaceTablet` and add assertion for those as well.

Reply:
Good idea - let me see what I can whip up here.